packages/eslint-plugin/tests/rules/no-confusing-void-expression.test.ts

```diff
@@ -473,5 +739,149 @@ function notcool(input: string) {
         },
       ],
     },
+    {
+      options: [{ ignoreVoidInVoid: true }],
+      code: `
+function test() {
+  return console.log('foo');
+}
+      `,
+      errors: [
+        {
+          line: 3,
+          column: 10,
+          messageId: 'invalidVoidExprReturnLast',
+        },
+      ],
+      output: `
+function test() {
+  console.log('foo');
+}
+      `,
+    },
+    {
+      options: [{ ignoreVoidInVoid: true }],
+      code: "const test = () => console.log('foo');",
+      errors: [
+        {
+          line: 1,
+          column: 20,
+          messageId: 'invalidVoidExprArrow',
+        },
+      ],
+      output: "const test = () => { console.log('foo'); };",
+    },
+    {
+      options: [{ ignoreVoidInVoid: true }],
+      code: `
+const test = () => {
+  return console.log('foo');
+};
+      `,
+      errors: [
+        {
+          line: 3,
+          column: 10,
+          messageId: 'invalidVoidExprReturnLast',
+        },
+      ],
+      output: `
+const test = () => {
+  console.log('foo');
+};
+      `,
+    },
+    {
+      options: [{ ignoreVoidInVoid: true }],
+      code: `
+function foo(): void {
+  const bar = () => {
+    return console.log();
+  };
+}
+      `,
+      errors: [
+        {
+          line: 4,
+          column: 12,
+          messageId: 'invalidVoidExprReturnLast',
+        },
+      ],
+      output: `
+function foo(): void {
+  const bar = () => {
+    console.log();
+  };
+}
+      `,
+    },
+    {
+      options: [{ ignoreVoidInVoid: true }],
+      code: `
+return console.log('foo');
+      `,
+      errors: [
+        {
+          line: 2,
+          column: 8,
+          messageId: 'invalidVoidExprReturn',
+        },
+      ],
+      output: `
+{ console.log('foo'); return; }
+      `,
+    },
+    {
+      options: [{ ignoreVoidInVoid: true }],
+      code: `
+      type Foo = () => void;
+      const test1: Foo = function () {
+        function testtt() {
+          return console.log(); // not reported - bug
```

Comment:
[Testing] I don't think we need these comments in tests because they might confuse people (the bug has already been fixed and this line is reported, but the comment says there is a bug)